pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -262,7 +262,7 @@ func validateServiceInstanceUpdate(instance *sc.ServiceInstance) field.ErrorList
 // to the spec to go through.
 func internalValidateServiceInstanceUpdateAllowed(new *sc.ServiceInstance, old *sc.ServiceInstance) field.ErrorList {
 	errors := field.ErrorList{}
-	if old.Generation != new.Generation && old.Status.ReconciledGeneration != old.Generation {
+	if old.Generation != new.Generation && old.Status.ReconciledGeneration != old.Generation && old.Status.CurrentOperation != "" {
```

Comment:
We should be able to remove the section about `old.Status.ReconciledGeneration != old.Generation`. We should never have the case where the current operation is set but the reconciled generation matches the generation. (Though looks like this is another gap in our validation functions)

Reply:
Ah, not a gap; found the check [here](https://github.com/kubernetes-incubator/service-catalog/blob/f86b8aaa4c5cf64e58717289b49e8898546a18d8/pkg/apis/servicecatalog/validation/instance.go#L244). So yes, this should be able to be removed safely.

Reply:
I think that I had in my head that there was an edge case where the current update could be changing the reconciled generation so that it matched the generation, which is why the check is for the old reconciled generation and old generation. But that doesn't make sense since we should only be updating *either* the spec *or* the status. I'll remove the unneeded part.